src/lib/web3-instance.ts

```diff
@@ -1,14 +1,15 @@
+import { ChainId } from '@shapeshiftoss/caip'
 import Web3 from 'web3'
 import { getWeb3Provider } from 'lib/web3-provider'
 
-let maybeWeb3Instance: Web3 | null
+const web3InstanceMap: Map<ChainId, Web3> = new Map()
```

Comment:
We now use a map of `web3` instances, as we need one for each RPC URL, which is approximately one per EVM chain.